reporthandling/apis/cloudmetadata.go

```diff
@@ -0,0 +1,41 @@
+package apis
+
+import "strings"
+
+type CloudProviderName string
+
+const (
+	GCP CloudProviderName = "GCP" // TODO: Why do we have this name ??
+	GKE CloudProviderName = "GKE"
+	EKS CloudProviderName = "EKS"
+	AKS CloudProviderName = "AKS"
+)
+
+type ICloudParser interface {
+
+	// Get the full and original name of the cluster. For example, for a cluster named "gke_project_zone_my-cluster" in GKE, the name is "gke_project_zone_my-cluster"
+	GetName() string
+
+	// Get provider name
+	Provider() CloudProviderName
+
+	// Parse the name of the cluster and return the prefix, suffix. For example, for a cluster named "gke_project_zone_my-cluster" in GKE, the prefix is "gke_project_zone" and the suffix is "my-cluster"
+	Parse() (string, string, error)
+
+	// // Parse the name of the cluster and return the prefix. For example, for a cluster named "gke_project_zone_my-cluster" in GKE, the prefix is "gke_project_zone"
```

Comment:
I commented out since I did not implement the functions listed bellow.
I believe we might use some of them in the future